drivers/sensor/apds9253/apds9253.c

```diff
@@ -121,9 +121,115 @@ static int apds9253_channel_get(const struct device *dev, enum sensor_channel ch
 	return 0;
 }
 
+static int apds9253_attr_set_gain(const struct device *dev, enum sensor_gain_apds9253 gain)
+{
+	const struct apds9253_config *config = dev->config;
+	struct apds9253_data *drv_data = dev->data;
+	uint8_t value;
+
+	if (drv_data->gain == gain) {
+		return 0;
+	}
+
+	switch (gain) {
+	case APDS9253_SENSOR_GAIN_1:
+		value = APDS9253_LS_GAIN_RANGE_1;
+		break;
+	case APDS9253_SENSOR_GAIN_3:
+		value = APDS9253_LS_GAIN_RANGE_3;
+		break;
+	case APDS9253_SENSOR_GAIN_6:
+		value = APDS9253_LS_GAIN_RANGE_6;
+		break;
+	case APDS9253_SENSOR_GAIN_9:
+		value = APDS9253_LS_GAIN_RANGE_9;
+		break;
+	case APDS9253_SENSOR_GAIN_18:
+		value = APDS9253_LS_GAIN_RANGE_18;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_LS_GAIN_REG, APDS9253_LS_GAIN_MASK,
+				   (value & APDS9253_LS_GAIN_MASK))) {
+		LOG_ERR("Not able to set light sensor gain is not set");
```

Comment:
```suggestion
		LOG_ERR("Not able to set light, sensor gain is not set");
```
Although this LOG_MSG is a bit unclear since only the gain is getting set over here, could you reframe this?